### PR TITLE
3.0.fix sync server settings

### PR DESCRIPTION
**Pull request by kalisp**
Updated Sync Server to use settings.

Better logging, documentation

### CODE REVIEW

File: pype/plugins/global/publish/integrate_new.py [hunk 1]
```diff
@@ -929,16 +929,25 @@ def prepare_file_info(self, path, size=None, file_hash=None, sites=None):
         Returns:
             rec: dictionary with filled info
         """
+        local_site = 'studio'  # default
+        remote_site = None
+        sync_server_presets = None
         try:
-            sync_server_presets = config.get_presets()["sync_server"]["config"]
+            settings = pype.api.get_current_project_settings()
+            sync_settings = settings.get("global")["sync_server"]
```
Comment (iLLiCiTiT) [inline pype/plugins/global/publish/integrate_new.py:937]: I think this can be `settings["global"]["sync_server"]`

File: pype/tools/settings/settings/gui_schemas/projects_schema/schema_main.json [hunk 2]
```diff
@@ -2,73 +2,71 @@
     "key": "project",
     "type": "dict",
     "children": [
+    {
```
Comment (iLLiCiTiT) [inline pype/tools/settings/settings/gui_schemas/projects_schema/schema_main.json:5]: Please do not mess up with indentantion it helps to see what is child of what...

File: pype/plugins/global/publish/integrate_new.py [hunk 3]
```diff
@@ -929,16 +929,25 @@ def prepare_file_info(self, path, size=None, file_hash=None, sites=None):
...
         except KeyError:
             log.debug(("There are not set presets for SyncServer."
                        " No credentials provided, no synching possible").
                       format(str(sync_server_presets)))
 
-        local_site = sync_server_presets.get("active_site", "studio").strip()
-        remote_site = sync_server_presets.get("remote_site")
-
         rec = {
             "_id": io.ObjectId(),
```
Comment (iLLiCiTiT) [inline pype/plugins/global/publish/integrate_new.py:952]: It's not part of this PR but could you remove `avalon.io` if only `ObjectId` is used? Use rather `from bson.objectid import ObjectId`.

File: pype/plugins/global/publish/integrate_new.py (hunk 1 above)
Comment (mkolar) [reply pype/plugins/global/publish/integrate_new.py:937]: I agree, probably better if it errors out on us, if the settings are missing.

File: pype/plugins/global/publish/integrate_new.py (hunk 3 above)
Comment (kalisp) [reply pype/plugins/global/publish/integrate_new.py:952]: There are another methods used out of avalon.io (find_one, update_one), but maybe `AvalonMongoDB` should be used instead of avalon.io? But that should be in different PR, I wouldn't mess with it here.

File: pype/tools/settings/settings/gui_schemas/projects_schema/schema_main.json (hunk 2 above)
Comment (kalisp) [reply pype/tools/settings/settings/gui_schemas/projects_schema/schema_main.json:5]: That's @mkolar formatting, I am guessing it comes from some plugin, we should decide what kind of formatting should be used or we will be moving it there and back again.

File: pype/plugins/global/publish/integrate_new.py (hunk 3 above)
Comment (iLLiCiTiT) [reply pype/plugins/global/publish/integrate_new.py:950]: Sorry, bad spot it was in Sync Server module.

**Comment by kalisp** [comment]
Implemented Module Manager.

Changed settings key to "sync_server" according to new PRs

**Review by mkolar — APPROVED** [review]
